### PR TITLE
Milo Gnav Multi Cloud Sign In

### PR DESCRIPTION
**Pull request by rgclayton**
Updated GNAV Sign In link to have a dropdown menu:
* Added dropdown to allow for multiple sign in options.
* Updated profile CSS for desktop and mobile experiences to be consistent with bussiness.adobe.com

Resolves: [MWPW-NUMBER](https://jira.corp.adobe.com/browse/MWPW-117663)

**Milo Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/rclayton/multi-cloud-signin?martech=off
- After: https://rclayton-gnav-multi-signin--milo--adobecom.hlx.page/drafts/rclayton/multi-cloud-signin?martech=off

**Bacom Test URLs:**
- Before: https://main--bacom--adobecom.hlx.page/drafts/rclayton/multi-cloud-signin?martech=off
- After: https://main--bacom--adobecom.hlx.page/drafts/rclayton/multi-cloud-signin?milolibs=rclayton-gnav-multi-signin&martech=off

image

### CODE REVIEW

**Comment by auniverseaway** [comment]
One thing I noticed is that our small menus (including this) have too much padding. You may want to review the spec on the zero height.

File: libs/blocks/gnav/gnav.js [hunk 1]
```diff
@@ -421,14 +421,29 @@ class Gnav {
   };
 
   decorateSignIn = (blockEl, profileEl) => {
+    const profileDropDown = blockEl.querySelector(':scope > div:nth-child(2)');
```
Comment (meganthecoder) [inline libs/blocks/gnav/gnav.js:424]: Is there a more specific selector available? It seems like relying on the order could make this fragile if the authoring isn't done exactly right

File: libs/blocks/gnav/gnav.js [hunk 2]
```diff
@@ -421,14 +421,29 @@ class Gnav {
...
-    profileEl.addEventListener('click', (e) => {
-      e.preventDefault();
-      window.adobeIMS.signIn();
-    });
+
+    if (profileDropDown) {
+      const id = `navmenu-${blockEl.className}`;
+
+      profileDropDown.id = id;
+      profileEl.classList.add('gnav-navitem', 'has-Menu');
```
Comment (meganthecoder) [inline libs/blocks/gnav/gnav.js:434]: Lowercase `has-menu`

**Review by meganthecoder — APPROVED** [review]
Looks good. Just a few suggestions

File: libs/blocks/gnav/gnav.js (hunk 2 above)
Comment (rgclayton) [reply libs/blocks/gnav/gnav.js:434]: @meganthecoder `has-Menu` is referenced in multiple places in the css.

https://github.com/adobecom/milo/blob/1fc340e1f10e929e539ca1df8fed6b460f20c20d/libs/blocks/gnav/gnav.css#L400

File: libs/blocks/gnav/gnav.js (hunk 1 above)
Comment (rgclayton) [reply libs/blocks/gnav/gnav.js:424]: Not with out writing a new one, based on this second child option.

**Comment by rgclayton** [comment]
> One thing I noticed is that our small menus (including this) have too much padding. You may want to review the spec on the zero height.

Will take a look and update.

File: libs/blocks/gnav/gnav.js (hunk 2 above)
Comment (meganthecoder) [reply libs/blocks/gnav/gnav.js:434]: Seems like a simple find and replace. Unless you're saying that the class comes from outside the Milo code. In that case, I understand that you can't change it

**Comment by rgclayton** [comment]
> One thing I noticed is that our small menus (including this) have too much padding. You may want to review the spec on the zero height.

@auniverseaway This has been fixed.

**Review by chrischrischris — APPROVED** [review]

File: libs/blocks/gnav/gnav.js (hunk 2 above)
Comment (auniverseaway) [reply libs/blocks/gnav/gnav.js]: This is a hold over from our pre style lint days. I'm fine with either fixing or leaving.

Comment (rgclayton) [reply libs/blocks/gnav/gnav.js]: I updated this.

**Comment by auniverseaway** [comment]
Screen Shot 2022-10-24 at 2 04 38 PM

I'm still seeing some extra padding compared to the Consonant spec.

Also, looks like there was some CLS introduced with this change. You can watch search pop to the left when sign-in in loads.

**Comment by auniverseaway** [comment]
Verified on [Stock](https://main--stock--adobecom.hlx.page/pages/artisthub/?milolibs=rclayton-gnav-multi-signin&martech=off) as well.

**Pull request merged**